### DISCUSSION TITLE
Clarifications on AI Observability Concepts for "Observability for AI Systems" executive challenge

### DISCUSSION THREAD

**Discussion by AntonioJesusGarciaPalomo** [msusdev/Innovation-Challenge#18] [Q&A]
I am working on a project that needs to align with the following challenge description:

> "Monitoring models in production is essential. Changes in data and behavior can influence your generative AI application over time, resulting in outdated systems that negatively affect business outcomes and expose organizations to risks. Show how to implement AI observability, including monitoring, tracking, and analyzing your AI model systems throughout their lifecycles. Ensure that systems operate effectively, ethically, and reliably by identifying issues like model drift, bias, performance degradation, and data quality problems."

To better address this challenge, I need clarification on the following concepts:

1. **Tracking**: 
   - What does the term "tracking" exactly mean in the context of AI observability? 
   - Does it refer solely to monitoring model performance metrics, or does it involve other elements such as data changes, operational logs, or user behavior?

2. **Model Drift vs. Performance Degradation**:
   - What is the distinction between "model drift" and "performance degradation"? 
   - Are they related phenomena, or do they represent completely separate issues?

3. **Data Quality Problems**:
   - When the challenge mentions "data quality problems," is it referring to issues in:
     - The data fed into the model during inference?
     - The data used for retraining the model?
   - Or does it encompass both types of data?

4. **Model Lifecycle**:
   - How is the lifecycle of an AI model determined?
   - Are there established frameworks or best practices for defining and tracking this lifecycle?
   - Where can I find resources or guidelines to better understand and implement lifecycle management for AI models?

These clarifications will help me ensure my project aligns closely with the challenge requirements. Any guidance, examples, or resources would be greatly appreciated.

**Comment by MatthewCalder-msft** [comment]
I’ll first try to answer your questions, and then I’ll provide an additional view of what the judges (and Azure customers) would find interesting. The goal is not to make a perfect solution that solves for everything, but to show what is possible.  

1.	Tracking in AI observability involves understanding the internal state of a system based on its external outputs, such as logs, metrics, traces, and other telemetry data. It helps developers monitor, troubleshoot, and debug complex systems by providing a comprehensive view of their behavior and performance. This is different from monitoring, which is more limited and focuses on specific metrics
2.	Model drift refers to the degradation of a machine learning model's performance due to changes in data or in the relationships between input and output variables. Performance degradation is a broader term that includes model drift but also covers other factors that affect the model's performance, such as changes in the system's environment or infrastructure
3.	Data quality covers both inference data and retraining data. There is an old phrase “Garbage in, garbage out” that I think of. The system might not have the right data to work with. 
4.	Solving for the model’s lifecycle might be beyond the scope of what can be done during the hackathon, but it would be good to mention in your presentation that you have thought about it. Best practices for tracking the model's lifecycle involve using frameworks like MLOps, which streamline the machine learning lifecycle from building models to deployment and management. This includes using tools for version control, continuous integration and deployment (CI/CD), and monitoring model performance and drift

As far as what a project might look like, below is a write up I got from a colleague a couple of months ago about the problem to solve for. Is that helpful?

Objective: Build an observability feature that tracks AI performance, detects hallucinations (incorrect or fabricated responses), and identifies effective usage patterns to ensure higher output quality and promote the full potential of AI.
Requirements:
•	Monitoring Metrics:
o	Implement tracking for key performance indicators like response time, success/error rates, and resource utilization.
•	Hallucination Detection Matrix:
o	Create a matrix that identifies and visualizes common hallucination patterns based on flagged or test responses.
o	Provide in-app feedback functionality for users to flag hallucinations, which feeds into the matrix for analysis.
•	Positive Usage Tracking:
o	Allow users to flag particularly effective or useful responses and analyze these flagged interactions to identify different and effective ways people are using the AI.
o	Utilize insights from positive usage tracking to educate users on the powerful and effective ways to leverage the AI tool.
